src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        addNode(value, size, tail, null);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> currentNode = getNode(index);
+            currentNode.prev = addNode(value, index, currentNode.prev, currentNode);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexExistence(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexExistence(index);
+        Node<T> currentNode = getNode(index);
+        T currentValue = currentNode.value;
+        currentNode.value = value;
+        return currentValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexExistence(index);
+        Node<T> currentNode = getNode(index);
+        unlink(currentNode);
+        return currentNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (currentNode.value == object || object != null && object.equals(currentNode.value)) {
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        if (currentNode != null) {
+            unlink(currentNode);
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getNode(int index) {
+        if (index < size / 2) {
+            Node<T> current = head;
+            int i = 0;
+            while (current != null && i <= index) {
+                if (i == index) {
+                    return current;
+                }
+                i++;
+                current = current.next;
+            }
+        } else {
+            Node<T> current = tail;
+            int i = size - 1;
+            while (current != null && i >= index) {
+                if (i == index) {
+                    return current;
+                }
+                i--;
+                current = current.prev;
+            }
+        }
+        return null;
+    }
+
+    private Node<T> addNode(T value, int index, Node<T> previousNode, Node<T> nextNode) {
```

Comment:
Any method should have as less arguments as possible. 0 or 1 is a good solution. 3 arguments is the maximum reasonable number according to Clean Code rules.
Delete this method and create two other:
private void addLast(T value)
private void addBefore(T value, Node<T> target)